CommandHandlerTesterLib/TestLongActionPlugin.cs

```diff
@@ -0,0 +1,52 @@
+﻿using CA_DataUploaderLib;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CommandHandlerTesterLib
+{
+    public class TestLongActionPlugin : LoopControlPlugin
+    {
+        public TestLongActionPlugin(CommandHandler cmd) : base(cmd)
+        {
+            AddCommand(nameof(LongAction), LongAction);
+            AddCommand("help", HelpMenu);
+        }
+
+        private bool HelpMenu(List<string> _)
+        { 
+            CALog.LogInfoAndConsoleLn(LogID.A, "longaction                - runs a bit longer action");
+            return true;
+        }
+
+        private bool LongAction(List<string> arg)
+        {
+            Task.Run(DoLongAction);
+            return true;
+        }
+
+        private async Task DoLongAction()
+        {
+            try
+            {
+                Console.WriteLine("Waiting for the next multiple of 10 value for IterationSensor");
+                var val = await WhenSensorValue("IterationSensor", v => v % 10 == 0, TimeSpan.FromSeconds(12));
+                Console.WriteLine($"IterationSensor = {val}. Waiting for value to increase by 3");
+                var vector = await When(e => e["IterationSensor"].Value >= val + 3, TimeSpan.FromSeconds(5)); // alt way of waiting / can target multiple sensor values too
```

Comment:
@copenhagenatomics I could add a Seconds convenience method to the base class, so we can do:
`await When(e => e["IterationSensor"].Value >= val + 3, Seconds(5))`
also
`await Task.Delay(Seconds(4));`
And perhaps a Delay method, so: `await Delay(Seconds(4))`

Reply:
I like this: await Delay(Seconds(4))

Reply:
@copenhagenatomics added PRs with this